pkg/k8s/volumes/crud.go

```diff
@@ -13,18 +13,77 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+const maxRetries = 300
+
 //Destroy destroys the volume claim for a given dev environment
 func Destroy(name string, dev *model.Dev, c *kubernetes.Clientset) error {
 	vClient := c.CoreV1().PersistentVolumeClaims(dev.Namespace)
 	log.Infof("destroying volume claim '%s'...", name)
-	for {
+
+	if err := checkIfAttached(name, dev, c); err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	for i := 0; i < maxRetries; i++ {
 		err := vClient.Delete(name, &metav1.DeleteOptions{})
 		if err != nil {
 			if strings.Contains(err.Error(), "not found") {
+				log.Infof("volume claim '%s' successfully destroyed", name)
 				return nil
 			}
+
 			return fmt.Errorf("error getting kubernetes volume claim: %s", err)
 		}
-		time.Sleep(1 * time.Second)
+
+		<-ticker.C
+		if i%10 == 5 {
+			log.Infof("waiting for volume claim '%s' to be destroyed...", name)
+		}
+	}
+
+	return fmt.Errorf("volume claim '%s' wasn't destroyed after 300s", name)
+}
+
+func checkIfAttached(pvc string, dev *model.Dev, c *kubernetes.Clientset) error {
```

Comment:
This code does not work with labels. I would check if pods are scheduled for termination:
https://github.com/okteto/okteto/blob/master/pkg/k8s/pods/pod.go#L64

Reply:
ah, good point. I'll check that way.